Neos.ContentRepository.Core/Classes/Infrastructure/Property/PropertyConverter.php

```diff
@@ -44,8 +44,8 @@ public function serializePropertyValues(
         $serializedPropertyValues = [];
 
         foreach ($propertyValuesToWrite->values as $propertyName => $propertyValue) {
-            if (!isset($nodeType->getProperties()[$propertyName]) && $propertyValue === null) {
-                // The property is undefined and the value null => we want to remove it, so we set it to null
+            if ($propertyValue === null) {
```

Comment:
`!isset($nodeType->getProperties()[$propertyName])` is at that point always `false` because we only allow to setProperties that actually exist in the current schema - thats why the the code path was previously never taken.